### ISSUE TITLE
[BUG] attention_mask rotary_pos_emb  and packed_seq_params  did not perform a shift for mtp model

### ISSUE THREAD

**Issue by OutisKwak** [NVIDIA/Megatron-LM#1675]
**Describe the bug**
As stated above， for mtp, we can see, input_ids labels and loss_mask call a roll_tensor function to perform shift.However, the attention_mask, rotary_pos_emb, and packed_seq_params should also perform the same operation.

https://github.com/NVIDIA/Megatron-LM/blob/5413c63a3733f4cbb30a7f16a45e03760b2a98cb/megatron/core/transformer/multi_token_prediction.py#L641

**To Reproduce**
Steps to reproduce the behavior. The easier it is to reproduce the faster it will get maintainer attention.

**Expected behavior**

**Stack trace/logs**
If applicable, add the stack trace or logs from the time of the error.

**Environment (please complete the following information):**
 - Megatron-LM commit ID
 - PyTorch version
 - CUDA version
 - NCCL version

**Proposed fix**
If you have a proposal for how to fix the issue state it here or link to a PR.

**Additional context**

**Comment by Skylion007** [comment]
There is an OP PR that fixes this, not sure why it isn't merged in. Flagging: @BoxiangW @ko3n1g https://github.com/NVIDIA/Megatron-LM/pull/1651

**Comment by BoxiangW** [comment]
@Skylion007 hi, in discussion https://github.com/NVIDIA/Megatron-LM/issues/1650#issuecomment-3021668915 the author of that PR mentioned that the results are the same. Do you think it applies for your case as well?

**Comment by BestJuly** [comment]
Hi @Suparjie, expect for the `rotaty_pos_embed`, why do you think the mask should perform a shift? In my understanding, after concatenation of hidden states and embeddings, the sequence can be treated as a new input sequence data, so we can also apply the causal mask to it, which will result in the same `attention_mask` and `packed_seq_params` in the calculation. 

For the shift of `rotaty_pos_embed`, I agree that it should be shifted but the result does return the same values. I think this is because the numeric difference is too little to be expressed in regular training data format.

**Comment by BoxiangW** [comment]
Closing, feel free to re-open if needed. Seems resolved
